### PR TITLE
Add @JsonProperty annotation to LeadControllerEntry & LeadControllerResponse

### PR DESCRIPTION
**Pull request by deepthi912**
This PR adds @JsonProperty annotation in the constructors of 2 responses(LeadControllerResponse & LeadControllerEntry).
It helps during the integration of this api responses to external systems to perform health check.

Note: I am doubting if there is a possibility more such response classes which might need annotation fixes similar to this. Worth checking this later

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java [hunk 1]
```diff
@@ -211,12 +212,16 @@ public List<String> getTableNames() {
 
   @JsonIgnoreProperties(ignoreUnknown = true)
   public static class LeadControllerResponse {
+    @JsonProperty("isLeadControllerResourceEnabled")
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java:215]: Can you verify if it should be `leadControllerResourceEnabled` or `isLeadControllerResourceEnabled`?
A JSON ser/de test will be very helpful to verify this and prevent breaking it in the future

Comment (deepthi912) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java]: Added a test to verify this and refactored previously added ser/deser tests into a new util test class.

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java [hunk 2]
```diff
@@ -211,18 +212,22 @@ public List<String> getTableNames() {
...
+    public LeadControllerResponse(
+        @JsonProperty("leadControllerResourceEnabled") boolean leadControllerResourceEnabled,
+        @JsonProperty("leadControllerEntryMap") Map<String, LeadControllerEntry> leadControllerEntryMap) {
+      _leadControllerResourceEnabled = leadControllerResourceEnabled;
       _leadControllerEntryMap = leadControllerEntryMap;
     }
 
-    public boolean isLeadControllerResourceEnabled() {
-      return _isLeadControllerResourceEnabled;
+    public boolean getLeadControllerResourceEnabled() {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java:229]: For `boolean` response type, we usually put `is` as the prefix of the method
```suggestion
    public boolean isLeadControllerResourceEnabled() {
```

File: pinot-controller/src/test/java/org/apache/pinot/controller/util/JsonSerializerTest.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    assertNotNull(tableResponseString);
+    JsonNode tableResponseJsonNode = JsonUtils.stringToJsonNode(tableResponseString);
+    assertTrue(tableResponseJsonNode.get("needReload").asBoolean());
+
+    JsonNode serversList = tableResponseJsonNode.get("serverToSegmentsCheckReloadList");
+    JsonNode serverResp = serversList.get("instance123");
+    assertEquals(serverResp.get("instanceId").asText(), "instance123");
+    assertTrue(serverResp.get("needReload").asBoolean());
+
+    assertEquals("{\n" + "  \"needReload\" : true,\n" + "  \"serverToSegmentsCheckReloadList\" : {\n"
```
Comment (Jackie-Jiang) [inline pinot-controller/src/test/java/org/apache/pinot/controller/util/JsonSerializerTest.java:66]: This is not very readable. You may disable the formatter by putting a comment `//@formatter:off` and enable it by putting `//@formatter:on` after the assert

Comment (deepthi912) [reply pinot-controller/src/test/java/org/apache/pinot/controller/util/JsonSerializerTest.java]: Refactored to a more readable one.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**